Tribler/Policies/BoostingManager.py

```diff
@@ -0,0 +1,676 @@
+        then insert it to our data
+        """
+        def compare_torrents(torrent_1, torrent_2):
+            """
+            comparing swarms. We don't want to download same swarm with different infohash
+            :return: whether those t1 and t2 similar enough
+            """
+            files1 = [files for files in torrent_1['metainfo'].get_files_with_length() if files[1] > 1024 * 1024]
+            files2 = [files for files in torrent_2['metainfo'].get_files_with_length() if files[1] > 1024 * 1024]
+
+            if len(files1) == len(files2):
+                for ft1 in files1:
+                    for ft2 in files2:
+                        if ft1[1] != ft2[1] or lev(ft1[0], ft2[0]) > 5:
+                            return False
+                return True
+            return False
+
+        # Remember where we got this torrent from
+
+        self._logger.debug("remember torrent %s from %s", torrent, source_to_string(source))
+
+        torrent['source'] = source_to_string(source)
+
+        boost_source = self.boosting_sources.get(source, None)
+        if not boost_source:
+            self._logger.info("Dropping torrent insert from removed source: %s", repr(torrent))
+            return
+        elif boost_source.archive:
+            torrent['preload'] = True
+            torrent['prio'] = 100
+
+        # If duplicates exist, set is_duplicate to True, except for the one with the most seeders.
+        duplicates = [other for other in self.torrents.values() if compare_torrents(torrent, other)]
+        if duplicates:
+            duplicates += [torrent]
+            healthiest_torrent = max([(torrent['num_seeders'], torrent) for torrent in duplicates])[1]
+            for duplicate in duplicates:
+                is_duplicate = healthiest_torrent != duplicate
+                duplicate['is_duplicate'] = is_duplicate
+                if is_duplicate and duplicate.get('download', None):
+                    self.stop_download(duplicate)
+
+        self.torrents[infohash] = torrent
+
+    def on_torrent_notify(self, subject, change_type, infohash):
+        """
+        Notify us when we have new seeder/leecher value in torrent from tracker
+        """
+        if infohash not in self.torrents:
+            return
+
+        self._logger.debug("infohash %s %s %s updated", subject, change_type, hexlify(infohash))
+
+        tdict = self.torrent_db.getTorrent(infohash, keys=['C.torrent_id', 'infohash', 'name',
+                                                           'length', 'category', 'status', 'num_seeders',
+                                                           'num_leechers'])
+
+        if tdict:
+            infohash_str = hexlify(tdict['infohash'])
+
+            new_seed = tdict['num_seeders']
+            new_leecher = tdict['num_leechers']
+
+            if new_seed - self.torrents[tdict['infohash']]['num_seeders'] \
```

Comment:
@ardhipoetra  I really don't see why your way is better what's the difference in behavior compared to @devos50 nicer proposal?